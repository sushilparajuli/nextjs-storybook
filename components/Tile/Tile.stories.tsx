import type { Meta, StoryObj } from "@storybook/react";
import { expect } from "@storybook/jest";
import { within } from "@storybook/testing-library";

import { Tile } from "./Tile";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta: Meta<typeof Tile> = {
  title: "Components/Tile",
  component: Tile,
  tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<typeof Tile>;

export const BasicTile: Story = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    await expect(canvas.getByRole("heading")).toBeInTheDocument();
  },
  args: {
    header: "Header",
    children: `Content goes here`,
  },
};
