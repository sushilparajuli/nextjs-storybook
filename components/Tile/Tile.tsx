import { FC, ReactNode } from "react";
import styled from "@emotion/styled";
import { borderRadius, boxShadow } from "@/components/styles";

export type Props = {
  /** Header */
  header: ReactNode;
  /** children */
  children: ReactNode;
};

export const Section = styled.section`
  padding: 1vmin 4vmin 4vmin;
  background: ${({ theme }) => theme.background};
  color: ${({ theme }) => theme.font.regular};
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)}
  ${borderRadius()}
`;

export const Tile: FC<Props> = ({ header, children, ...rest }) => (
  <>
    <Section {...rest}>
      <h2>{header}</h2>
      {children}
    </Section>
  </>
);
