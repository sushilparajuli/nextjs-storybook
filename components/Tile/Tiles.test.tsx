import { render, screen } from "@/test-utils";
import { Tile } from "./Tile";
describe("Tile test cases", () => {
  it("should render", () => {
    const { asFragment } = render(
      <Tile header="This is title">This is content</Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
