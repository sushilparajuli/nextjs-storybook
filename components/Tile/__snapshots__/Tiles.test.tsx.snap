// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render 1`] = `
<DocumentFragment>
  <section
    class="css-1anqmly"
  >
    <h2>
      This is title
    </h2>
    This is content
  </section>
</DocumentFragment>
`;

exports[`Tile test cases should render 1`] = `
<DocumentFragment>
  <section
    class="css-1anqmly"
  >
    <h2>
      This is title
    </h2>
    This is content
  </section>
</DocumentFragment>
`;
